pkg/services/service.go

```diff
@@ -69,8 +69,7 @@ func (e *Engine) Go(fn func(context.Context)) {
 	e.wg.Add(1)
 	go func() {
 		defer e.wg.Done()
-		ctx, cancel := e.StopChan.NewCtx()
-		defer cancel()
+		ctx, _ := e.StopChan.NewCtx()
```

Comment:
`StopChan` handles calling `cancel()`

Reply:
The parent cancellation is different. We also want to cancel this instance when the goroutine is complete. What are we trying to achieve here?